src/main/java/org/araymond/joal/core/ttorrent/client/announcer/request/AnnouncerExecutor.java

```diff
@@ -50,9 +50,10 @@ public void execute(final AnnounceRequest request) {
             }
         };
 
+        final Future<?> taskFuture = this.executorService.submit(task);
         this.currentlyRunning.put(
                 request.getAnnouncer().getTorrentInfoHash(),
-                new AnnouncerWithFuture(request.getAnnouncer(), this.executorService.submit(task))
+                new AnnouncerWithFuture(request.getAnnouncer(), taskFuture)

```

Comment:
Again, reverting back to how it was. This extra verbosity helps with readability IMHO.